### PR TITLE
Coarsen fields

### PR DESCRIPTION
**Pull request by RemiLehe**
This reduces the size of the regular plotfile output by coarsening the field. (i.e. averaging in all dimensions)

### CODE REVIEW

**Comment by WeiqunZhang** [comment]
In `coarsenCellCenteredFields`, `MultiFab small_mf` is a local variable.  It will be deleted at the end of each loop.  So `output_mf` will contain invalid pointers.  The function should take `Vector<MultiFab>& output_mf` instead.

**Comment by RemiLehe** [comment]
@WeiqunZhang Thanks for catching that! I should have tested the code after my latest modifications!

Ideally, I would like to avoid copying `mf_avg` in the case where field coarsening is not required (this is the reason why I used `Vector<MultiFab*>`) (see the file `WarpXIO.cpp`). My understanding is that switching to `Vector<MultiFab>` would require copying the MultiFabs. 
Any suggestion on how to do this in an elegant manner?

**Comment by WeiqunZhang** [comment]
I don't think you need to copy the data in case coarsening is not needed.  You are creating a Vector of MultiFab const* independent of coarsening or not.

**Comment by RemiLehe** [comment]
@WeiqunZhang I updated the PR according to your comment and tested the code on the Langmuir wave problem:

**- plot_coarsening_ratio = 1**
Screen Shot 2019-03-19 at 10 41 51 PM

**-plot_coarsening_factor=2**
Screen Shot 2019-03-19 at 10 39 57 PM

**-plot_coarsening_factor=4**
Screen Shot 2019-03-19 at 10 41 27 PM

File: Docs/source/running_cpp/parameters.rst [hunk 1]
```diff
@@ -483,6 +483,11 @@ Diagnostics and output
     Only used when mesh refinement is activated and ``warpx.plot_raw_fields`` is ``1``.
     Whether to output the data of the coarse patch, in the plot files.
 
+* ``warpx.plot_coarsening_ratio`` (`int`)
+    Reduce size of the field output by this ratio in each dimension.
+    (This is done by averaging the field.) The ratio should be a integer 
+    power of 2 (e.g. 2, 4, 8, ...).
+
```
Comment (WeiqunZhang) [inline Docs/source/running_cpp/parameters.rst]: Could we add the default is 1?

**Pull request merged**